src/Paths/ListFiles.ts

```diff
@@ -1,6 +1,27 @@
-import { BaseServer } from "../Server/BaseServer";
-import { root } from "./Directory";
-import { ContentFileMap, allContentFiles } from "./ContentFile";
+import type { BaseServer } from "../Server/BaseServer";
+import type { ContentFile, ContentFileMap, ContentFilePath } from "./ContentFile";
+import { type Directory, root } from "./Directory";
+import type { FilePath } from "./FilePath";
+
+/** Generator function to allow iterating through all content files on a server */
+export function* allContentFiles(server: BaseServer): Generator<[ContentFilePath, ContentFile], void, undefined> {
```

Comment:
I moved these here, to avoid imports from outside `Paths/` (to prevent recursive imports of stuff that goes in the vendor bundle). But, it turns out we only need type imports of those things, so this bit could be dropped.